### PR TITLE
In game server selector

### LINKED ISSUE #344: Create server select button
We now have a server hosted at https://central-europe-testing.curseofmirra.com/ so we need a button like we have in CoM where we select which server we want to connect to (localhost or europe)

### PR DESCRIPTION
**Pull request by tkz00**
Closes #344 

## Motivation

A runtime way to change the server to be connected is needed, in production it will serve the user to play on different servers, but the feature is introduced now since it's useful for testing during the development, specially now the mobile build is close, since previous to this feature to test different servers on a phone you needed to edit source code an rebuild the app, which takes an enormous amount of time.

## Summary of changes

Added a server button on the overworld scene, which describes the current server the user is corrected, selecting this button opens up a popup which let's you choose between localhost and europe servers, also letting you introduce a custom server domain and choose that to connect. All this functionality is handled by the `SelectServer.cs` script.

Consider now a `userId` is saved in player prefs for each server, so starting in local, changing to Europe and coming back to local will maintain you original local user (the user won't be shared between Europe and local), to stop this you can delete the user from the database or clear up the player prefs from the Unity editor, the second way is simpler but you delete the user id's from all the servers, you can't just delete one.

## How has this been tested?

Start the application and try to change servers, consider currently the only "production" server available is the Europe one, so you need at least one local server running, verify no exceptions appear and your user is created/brought from the backend. If possible also try to connect to your local server exposing the address through ngrok via the custom server address.

### CODE REVIEW

**Review by ncontinanza — CHANGES_REQUESTED** [review]
The problem of registering the user is solved (that was because of the seeds not being run in the backend), but I'm still getting several errors when trying to run a battle or summon a unit.

https://github.com/lambdaclass/champions_of_mirra/assets/17294394/93b061ac-0554-475c-9856-93fbea6ac5d1

**Comment by tkz00** [comment]
> The problem of registering the user is solved (that was because of the seeds not being run in the backend), but I'm still getting several errors when trying to run a battle or summon a unit.
> Screen.Recording.2024-04-29.at.14.02.44.mov

Nice catch Nico! This has been addressed!

File: client/Assets/Scripts/BackendConnection/ServerSelect.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+		if(string.IsNullOrEmpty(ServerSelect.Name) || string.IsNullOrEmpty(ServerSelect.Domain)) {
+			#if UNITY_EDITOR
+				ServerSelect.Name = "LOCALHOST";
+				ServerSelect.Domain = servers["LOCALHOST"];
+			#else
+				ServerSelect.Name = "EUROPE";
+				ServerSelect.Domain = servers["EUROPE"];
+			#endif
+			// ServerSelect.Name = "EUROPE";
+			// ServerSelect.Domain = servers["EUROPE"];
```
Comment (ncontinanza) [inline client/Assets/Scripts/BackendConnection/ServerSelect.cs:33]:
```suggestion
```

**Review by ncontinanza — CHANGES_REQUESTED** [review]

**Review by lotuuu — APPROVED** [review]

**Comment by lotuuu** [comment]
Later on we might want to put the Server Select option before actually creating connections, to prevent possible bugs

**Pull request merged**